app/src/main/java/com/example/myfitnessbuddy/adapters/BuddyAdapter.java

```diff
@@ -89,12 +92,13 @@ public void onClick(View v) {
             imageViewChat.setOnClickListener(new View.OnClickListener() {
```

Comment:
setOnClickListener cel mai bine se face in onCreateViewHolder din adapter. pentru ca onBind se cheama de fiecare data cand un item iese din screen si reintra, pe cand onCreateViewHolder se face doar o data. daca ai destul de multe date, va fi un lag cand se face scroll din cauza asta :) 

poti sa faci in onCreateViewHolder, si iei MatchedBuddy care corespunde acum cu getAdapterPosition, care iti va da pozitia din adapter pe care ai dat click.

si logica de startActivity ar fi bine sa fie chemata din activity / fragment, asa ca ai putea sa faci un listener pe care sa il dai parametru cand initializezi BuddyAdapter, si in onCreateViewHolder sa faci doar ceva de genul : 
`viewHolder.imageViewChat.setOnClickListener(listener.buddyClicked(holder.getAdapterPosition()))`

uite aici mai multe informatii despre asta : https://stackoverflow.com/questions/33845846/why-is-adding-an-onclicklistener-inside-onbindviewholder-of-a-recyclerview-adapt